hoodie-client/src/main/java/com/uber/hoodie/HoodieWriteClient.java

```diff
@@ -1273,14 +1275,15 @@ private void updateMetadataAndRollingStats(String actionType, HoodieCommitMetada
       // 2. Now, first read the existing rolling stats and merge with the result of current metadata.
 
       // Need to do this on every commit (delta or commit) to support COW and MOR.
-      for (WriteStatus status : writeStatusList) {
-        HoodieWriteStat stat = status.getStat();
+
+      for (HoodieWriteStat stat : writeStats) {
+        String partitionPath = stat.getPartitionPath();
         //TODO: why is stat.getPartitionPath() null at times here.
-        metadata.addWriteStat(status.getPartitionPath(), stat);
+        metadata.addWriteStat(partitionPath, stat);
```

Comment:
I remember running into some weird test fail when getting the partition path from the stat?